### ISSUE TITLE
Show not being picked up/populated correctly

### ISSUE THREAD

**Issue by Anaerin** [mrphlip/lrrbot#1518]
For a while now, the show value in lrrbot has just been flat-out wrong. I've been looking over the code and frankly I can't tell where it's being set automatically (or not). Either way, it's not doing it correctly, so I propose an option.

* The "shows" DB table gets a new column to store a RegEx for that show's name.
* When the Twitch title is updated, lrrbot tries each RegEx on the title 'til it gets a match, and updates the show value to match.
* If there are multiple matches, or no matches, the show remains unchanged so a moderator can override.

Also: If the show is overridden and the stream has been down for some time (an hour+?) the override is automatically removed.

Thoughts?

**Comment by Anaerin** [comment]
Oh, I found where I think the show is being set, which appears to be via the web API, possibly from LRR's overlay.

**Comment by mrphlip** [comment]
Yeah, when this system was originally built, we went with having the show info get [fed in from the LRR overlay](https://github.com/mrphlip/lrrbot/blob/master/www/api_v2.py#L78), since it seemed more reliable... back then, the show name in the stream title wasn't always in the same format, wasn't always updated properly, and was occasionally messed with for jokes... while the overlay always knows what show it is supposed to be.

But then 2020 happened and people started streaming from home, and the Xsplit overlay they used to use got replaced with an OBS overlay for people to use from home... and that new overlay didn't push the show to lrrbot, so it wouldn't update. And even now that they're mostly back in the office, they're still using OBS a lot more, so it's still not working.

Ultimately it's something we've known is a problem for a while, but it really hasn't been a priority to fix it.
